pymeasure/instruments/agilent/agilentL4534A.py

```diff
@@ -0,0 +1,333 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2023 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import numpy as np
+from numpy import typing
+
+from decimal import Decimal
+
+from pymeasure.errors import Error
+from pymeasure.instruments import Instrument, Channel
+from pymeasure.instruments.validators import strict_discrete_set, strict_range, strict_discrete_range
+from pymeasure.units import ureg
+
+
+def sample_count_function(value, values): return strict_discrete_range(value, values, 4)
+
+SAMPLE_COUNT_VALUES = [8, 128e6]
+PRE_TRIG_VALUES = [0, 128e6]
+RECORD_COUNT_VALUES = [1, 1024]
+
+TRIGGER_HOLDOFF_VALUES = ureg.Quantity(np.asarray([0, 10]), ureg.s)
+TRIGGER_DELAY_VALUES = ureg.Quantity([0, 3600], ureg.s)
+
+def _get_channel_config_process(values):
+    result = {
+        'range': ureg.Quantity(Decimal(values[0]), ureg.V),
```

Comment:
Do you get any benefit by using Decimal over a plain float? Especially, after we received a string and turned that into a float.